core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,288 @@
+   */
+  async createExploration(): Promise<void> {
+    await this.clickOn(createExplorationButton);
+    await this.clickOn(dismissWelcomeModalSelector);
+  }
+
+  /**
+   * Function to add content to a card.
+   * @param {string} questionText - The content to be added to the card.
+   */
+  async updateCardContent(questionText: string): Promise<void> {
+    await this.clickOn(stateEditSelector);
+    await this.waitForElementToBeClickable(stateContentInputField);
+    await this.page.click(stateContentInputField, {clickCount: 3});
+    await this.page.keyboard.press('Backspace');
+    await this.type(stateContentInputField, `${questionText}`);
+    await this.clickOn(saveContentButton);
+    await this.page.waitForSelector(stateContentInputField, {hidden: true});
+  }
+
+  /**
+   * Function to add an interaction to the exploration.
+   * @param {string} interactionToAdd - The interaction type to add to the Exploration.
+   */
+  async addInteraction(interactionToAdd: string): Promise<void> {
+    await this.clickOn(addInteractionButton);
+    await this.clickOn(interactionToAdd);
+    await this.clickOn(saveInteractionButton);
+  }
+
+  /**
+   * Function to display the Oppia responses section.
+   */
+  async viewOppiaResponses(): Promise<void> {
+    await this.page.waitForSelector(stateResponsesSelector, {visible: true});
+    await this.page.click(stateResponsesSelector);
+  }
+
+  /**
+   * Function to select the card that learners will be directed to from the current card.
+   */
+  async oppiaDirectlearnersTo(card: string): Promise<void> {
+    await this.clickOn(openOutcomeDestButton);
+    await this.waitForElementToBeClickable(destinationCardSelector);
+    await this.page.select(destinationCardSelector, card);
+  }
+
+  /**
+   * Function to name a new card in the exploration.
+   */
+  async nameNewCard(cardName: string): Promise<void> {
+    await this.page.type(addStateInput, cardName);
+    await this.waitForElementToBeClickable(saveOutcomeDestButton);
+    await this.page.click(saveOutcomeDestButton);
+  }
+
+  /**
+   * Function to save an exploration draft.
+   */
+  async saveExplorationDraft(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      const element = await this.page.$(mobileNavbarOptions);
+      if (!element) {
+        await this.clickOn(mobileOptionsButton);
+      }
+      await this.clickOn(mobileSaveChangesButton);
+      await this.clickOn(saveDraftButton);
+      await this.page.waitForNetworkIdle();
+    } else {
+      await this.clickOn(saveChangesButton);
+      await this.clickOn(saveDraftButton);
+      await this.page.waitForNetworkIdle();
+    }
+  }
+
+  /**
+   * Function to navigate to a specific card in the exploration.
+   * @param {string} cardName - The name of the card to navigate to.
+   */
+  async navigateToCard(cardName: string): Promise<void> {
+    if (cardName === 'Introduction' && this.isViewportAtMobileWidth()) {
```

Comment:
This won't do, you need to make this function more general and not assume a specific format of the exploration. It should error if the name isn't in the graph and generically navigate to the right state otherwise.

Reply:
Yes, the previous implementation was kinda vague and limited. I have tried to generalise it. The function now receives two parameters: one is the list of card names, and the other is the specific card to navigate to.